src/System.Management.Automation/engine/CommandCompletion/CompletionCompleters.cs

```diff
@@ -4383,6 +4383,12 @@ internal static IEnumerable<CompletionResult> CompleteFilename(CompletionContext
                         if (CompletionRequiresQuotes(completionText, !useLiteralPath))
                         {
                             var quoteInUse = quote == string.Empty ? "'" : quote;
+
```

Comment:
should this change be behind the experimental feature as well?

Reply:
This was a condition of the committee

Reply:
Except filename completion will now also escape `*` and `?` in non-literal path, the behaviour should be identical unless the experimental feature (the breaking part) is enabled.